plugins/inputs/docker/README.md

```diff
@@ -43,12 +43,27 @@ to gather stats from the [Engine API](https://docs.docker.com/engine/api/v1.24/)
   ## Timeout for docker list, info, and stats commands
   timeout = "5s"
 
-  ## Whether to report for each container per-device blkio (8:0, 8:1...) and
-  ## network (eth0, eth1, ...) stats or not
+  ## Whether to report for each container per-device blkio (8:0, 8:1...),
+  ## network (eth0, eth1, ...) and cpu (cpu0, cpu1, ...) stats or not.
+  ## Usage of this setting is discouraged since it will be deprecated in favor of 'perdevice_include'.
+  ## Default value is 'true' for backwards compatibility, please set it to 'false' so that 'perdevice_include' setting 
+  ## is honored.
   perdevice = true
-
-  ## Whether to report for each container total blkio and network stats or not
+  
+  ## Specifies for which classes a per-device metric should be issued
+  ## Possible values are 'cpu' (cpu0, cpu1, ...), 'blkio' (8:0, 8:1, ...) and 'network' (eth0, eth1, ...)
+  perdevice_include = ["cpu"]
```

Comment:
You should comment that line (as it is the default) and maybe add
`This setting has no effect if perdevice is set to true.`

Reply:
> You should comment that line (as it is the default)

It is the default only because I've set this value in the default config, but I am not defaulting to this value in code. 

If I default to it in code (i.e: on `Init()`), I'm thinking I should set `omitempty` on its toml annotation so that I am able to differentiate an intentional empty value (`perdevice_include=[]`, user does not want any perdevice metric) from a missing configuration item (this could happen if the user is using an old config with a new Telegraf version and doesn't have this new property in it; in this case, I should default to the backwards compatible value)

Does this make sense?


> and maybe add `This setting has no effect if perdevice is set to true.`
:+1:

Reply:
Well what should the default be to keep the previous behavior? That's up to you to decide. :-) Anyway, the option should be commented with the default used in code.

However, I guess 'omitemty' is only available for _encoding_ but not decoding toml. The easiest way is to specify the default when constructing the `Docker` instance (i.e. usually in `init()`). If you set `["cpu"]` there, you can be sure that the user was overriding the option if the field is empty in `Init()`. I would not do any magic here and set the default in `init()` and use what we get in `Init()`...

Reply:
Please make this the default in code and comment this line. IIRC this is what you need to reproduce the past behavior even if `perdevice` is `true`.

Reply:
You're right! didn't realize `omitempty` was only for encoding purposes. I have implemented the `init()` approach as suggested